### PR TITLE
Add support for stable goldens

### PR DESCRIPTION
**Pull request by DanTup**
This is to help with #548. There's now a `--stable` flag you can pass to `update-goldens` and also if you set `-DGOLDENS_SUFFIX` in `DART_VM_OPTIONS` it'll use them at runtime (currently no code does this, but I'll be rebasing #548 on top of this to use it soon).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -73,6 +73,13 @@ file output still looks reasonable and execute the following command to update t
 ./tool/update_goldens.sh
 ```
 
+To update the Stable versions of the Golden files, switch to Flutter's current stable
```
Comment (jacob314) [inline CONTRIBUTING.md]: Lets add a TODO somewhere to verify you really are on stable when you say you are.

Comment (DanTup) [reply CONTRIBUTING.md]: Will do! I wasn't sure how we could reliably do that (without fetching from GitHub) and figured the build will fail on the PR (once we get the stable tests running) so it's not critical. I'll a TODO though.

File: tool/update_goldens.sh [hunk 2]
```diff
@@ -4,9 +4,14 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-export DART_VM_OPTIONS="-DUPDATE_GOLDENS=true"
+if [ $1 = "--stable" ]; then
```
Comment (jacob314) [inline tool/update_goldens.sh]: Lets verify you are on stable if you claim you are. E.g. 
```
flutter --version
Flutter 1.5.2-pre.9 • channel unknown • unknown source
```
indicates I am not on stable.

File: CONTRIBUTING.md (hunk 1 above)
Comment (DanTup) [reply CONTRIBUTING.md]: (Done! I added two, since we should probably do the same for ToT too).

**Comment by DanTup** [comment]
Ok, I can try that. Being on the stable channel doesn't guarantee it's the right code (esp. if you're on a fork and you didn't keep your branches up-to-date), but I think it'll help catch the simple mistakes where we just didn't change branch.

**Comment by DanTup** [comment]
@jacob314 how's 6e7225726cfa15345f80cf26f28c8207d4f71899?

**Comment by jacob314** [comment]
lgtm
That looks good. Thanks for fixing this!

File: tool/update_goldens.sh [hunk 3]
```diff
@@ -4,9 +4,19 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-export DART_VM_OPTIONS="-DUPDATE_GOLDENS=true"
+echo "Checking flutter version..."
+if flutter --version | tee /dev/tty | grep -q 'channel stable'; then
+	export DART_VM_OPTIONS="-DUPDATE_GOLDENS=true -DGOLDENS_SUFFIX=_stable"
```
Comment (natebosch) [inline tool/update_goldens.sh]: Unless we have significant reason to diverge from published google style can we indent with 2 spaces here?

https://google.github.io/styleguide/shell.xml#Indentation

Comment (jacob314) [reply tool/update_goldens.sh]: Thanks for improving our shell style guide conformance! :)

Comment (DanTup) [reply tool/update_goldens.sh]: No good reason (except that I haven't read the style guides, and my VS Code defaults everything except Dart to tabs 🙃)

Fixed!

**Comment by jacob314** [comment]
lgtm

**Pull request merged**